src/libAtomVM/term.c

```diff
@@ -499,14 +513,52 @@ TermCompareResult term_compare(term t, term other, TermCompareOpts opts, GlobalC
             break;
 
         } else if (term_is_reference(t) && term_is_reference(other)) {
-            int64_t t_ticks = term_to_ref_ticks(t);
-            int64_t other_ticks = term_to_ref_ticks(other);
-            if (t_ticks == other_ticks) {
-                CMP_POP_AND_CONTINUE();
+            if (!term_is_external(t) && !term_is_external(other)) {
+                int64_t t_ticks = term_to_ref_ticks(t);
+                int64_t other_ticks = term_to_ref_ticks(other);
+                if (t_ticks == other_ticks) {
+                    CMP_POP_AND_CONTINUE();
+                } else {
+                    result = (t_ticks > other_ticks) ? TermGreaterThan : TermLessThan;
+                    break;
+                }
+            } else if (term_is_external(t) && term_is_external(other)) {
+                term node = term_get_external_node(t);
+                term other_node = term_get_external_node(other);
+                if (node == other_node) {
+                    uint32_t creation = term_get_external_node_creation(t);
+                    uint32_t other_creation = term_get_external_node_creation(other);
+                    if (creation == other_creation) {
+                        uint32_t len = term_get_external_reference_len(t);
+                        uint32_t other_len = term_get_external_reference_len(other);
+                        if (len == other_len) {
+                            uint32_t data[len];
+                            uint32_t other_data[len];
+                            term_get_external_reference_words(t, data);
+                            term_get_external_reference_words(other, other_data);
+                            int cmp = memcmp(data, other_data, len * sizeof(uint32_t));
```

Comment:
We might just get a pointer to the internal reference words so we don't need to copy them around.
The same applies to most of the other places where we briefly need to access those words.

Reply:
On 64 bits platforms, it means casting a `uint64_t*` pointer to `uint32_t*`. Do we want to do that?